sklearn/datasets/_samples_generator.py

```diff
@@ -1646,10 +1654,21 @@ def make_sparse_spd_matrix(
     """
     random_state = check_random_state(random_state)
 
-    chol = -sp.eye(dim)
+    # TODO: remove in 1.5
+    if dim != "deprecated":
+        warnings.warn(
+            (
+                "dim was deprecated in version 1.4 and will be removed in 1.5."
+                "Please use ``n_dim`` instead."
+            ),
+            FutureWarning,
+        )
+        n_dim = dim
```

Comment:
do not change `n_dim` but `_n_dim`. We should raise an error if a user try to specify both `n_dim` and `dim`.

I assume that the time of the deprecation, we need to have `n_dim=None` to detect explicitely if a user pass `n_dim=1` or not. This is annoying but I don't see how to detect this case otherwise. This would not change the default value because `None` would default to 1. Somehow, we can omit to document it explicitely, and make a more silent deprecation because I don't think that any user will use `n_dim=Noe` explicitely to set it to 1.